src/coreComponents/physicsSolvers/NonlinearSolverParameters.cpp

```diff
@@ -54,6 +54,11 @@ NonlinearSolverParameters::NonlinearSolverParameters( string const & name,
     setDescription( "Line search cut factor. For instance, a value of 0.5 will result in the effective application of"
                     " the last solution by a factor of (0.5, 0.25, 0.125, ...)" );
 
+  registerWrapper( viewKeysStruct::lineSearchStartingIterationString(), &m_lineSearchStartingIteration ).
+    setApplyDefaultValue( 0 ).
+    setInputFlag( InputFlags::OPTIONAL ).
+    setDescription( "Iteration when line search starts." );
```

Comment:
I am not sure I understand what this is.

Reply:
The idea is to delay line search application for several Newtown iterations
It's quite common that flow residual is not decreasing at the beginning of Newtown iterations and doing line search is not very effective there